data-service/src/main/java/com/amazonaws/blox/dataservice/model/InstanceGroup.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.model;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+import java.util.Set;
+
+@Value
+@Builder
+public class InstanceGroup {
+
+  @NonNull private String clusterArn;
+  private Set<Attribute> attributes;
```

Comment:
How would this work? Today, the only instance grouping criteria available in the ECS API is through the Cluster Query Language, which is what the [`listContainerInstances`'s `filter` argument accepts](http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ListContainerInstances.html#ECS-ListContainerInstances-request-filter).